python/lsst/daf/butler/cli/utils.py

```diff
@@ -667,3 +670,50 @@ def getFrom(ctx):
             return ctx.obj
         ctx.obj = MWCtxObj()
         return ctx.obj
+
+
+def yaml_presets(ctx, param, value):
+    """Click callback that reads additional values from the supplied
+    YAML file.
+
+    Parameters
+    ----------
+    ctx : `click.context`
+        The context for the click operation. Used to extract the subcommand
+        name.
+    param : `str`
+        The parameter name.
+    value : `object`
+        The value of the parameter.
+    """
+    ctx.default_map = ctx.default_map or {}
+    cmd_name = ctx.info_name
```

Comment:
`Context.info_name` is weird: when used with a command that is in a Group, as our subcommands are, then it contains the name of the subcommand, and so this works. But if you were to use this callback with the top level command (`butler` or `pipetask`) then it would contain the name of the script, including any extension (`.py`). Currently this would be ok for our commands because the butler script is called `butler` (not `butler.py`), and similar for `pipetask`. But it might pop up with other scripts if they don't omit the extension. Maybe it's worth stripping `.py` off the end here if it exists, or warning if `.py` is found? Or at least it's something to be aware of, someone may hit it someday.

Reply:
This effectively becomes the key to use in the dict read from the yaml file. Things are all fine so long as this cmd_name is itself mentioned in the yaml file. Given that we are only using this for subcommands I'm inclined to leave it as is at the moment and let things break.